### PR TITLE
security: add permissions block to workflows

### PR DESCRIPTION
**Pull request by reakaleek**
## Details

⚠️ This PR was created by an automated tool. Please review the changes carefully. ⚠️ 

We want to set the default permissions for workflows to read-only for contents. 
This is a security measure to prevent accidental changes to the repository.

This change adds a top-level permissions block to all workflows in the .github/workflows directory.
```yaml
permissions:
  contents: read
```

In some cases workflows might need more permissions than just `contents: read`.
Please checkout this branch and add the necessary permissions to the workflows.

If your workflow uses a Personal Access Token (PAT), we can still add the permissions block,
   but it will not have any effect.

Merging this PR as is might cause workflows that need more permissions to fail.

If there are any questions, please reach out to the @elastic/observablt-ci

### CODE REVIEW

**Review by xrmx — DISMISSED** [review]

File: .github/workflows/addToProject.yml [hunk 1]
```diff
@@ -6,6 +6,9 @@ on:
 env:
   MY_GITHUB_TOKEN: ${{ secrets.APM_TECH_USER_TOKEN }}
 
+permissions:
+  contents: read
```
Comment (v1v) [inline .github/workflows/addToProject.yml:10]: I'm not sure if `elastic/assign-one-project-github-action@1.2.2` requires some write access

Comment (basepi) [reply .github/workflows/addToProject.yml:10]: @AlexanderWert it looks like this github action is assigning our issues to the `OLD` apm-agents project anyway. Should it be disabled or it is still useful?

File: .github/workflows/test-reporter.yml [hunk 2]
```diff
@@ -8,6 +8,10 @@ on:
     types:
       - completed
 
+permissions:
+  contents: read
+  actions: read
```
Comment (v1v) [inline .github/workflows/test-reporter.yml:13]:
```suggestion
  actions: read
  checks: write
```

**Comment by xrmx** [comment]
The is a draft PR reworking the labeler workflow that is adding correct permissions too https://github.com/elastic/apm-agent-python/pull/2003

File: .github/workflows/labeler.yml [hunk 3]
```diff
@@ -6,6 +6,9 @@ on:
     types: [opened]
 env:
   MY_GITHUB_TOKEN: ${{ secrets.APM_TECH_USER_TOKEN }}
+permissions:
+  contents: read
```
Comment (reakaleek) [inline .github/workflows/labeler.yml:10]:
```suggestion
  contents: read
  issues: write
  pull-request: write
```

**Review by v1v — APPROVED** [review]

File: .github/workflows/addToProject.yml [hunk 4]
```diff
@@ -6,6 +6,9 @@ on:
...
+
```
Comment (reakaleek) [inline .github/workflows/addToProject.yml:11]:
```suggestion
```

File: .github/workflows/addToProject.yml (hunk 1 above)
Comment (reakaleek) [reply .github/workflows/addToProject.yml]: Removed in https://github.com/elastic/apm-agent-python/pull/1972/commits/febb916fd524cc22020a11cb5b0cafd8c6038e44

**Review by xrmx — APPROVED** [review]

**Pull request merged**